doc/rst/developer/chips/23.rst

```diff
@@ -0,0 +1,614 @@
+        }
+      }
+
+      // A contrived variation of DefaultRectangularDom
+      class MyClass3 : MyClass2 {
+        var ranges : rank * range(idxType);
+
+        // ***** Unable to initialize omitted field *****
+        proc init(param _rank, type _idxType) {
+          super.init(_rank, _idxType);
+        }
+
+        // ***** Unable to type check initialization *****
+        proc init(_ranges) {
+          ranges = _ranges;
+
+          super.init(_ranges.size, _ranges(1).idxType);
+        }
+      }
+
+   but can also be a problem for non-generic classes e.g.
+
+   .. code-block:: chapel
+
+      class C1 {
+        var D = { 1 .. n };
+        var A : [D] real;
+
+        proc init(_D) {
+          D = _D;
+          super.init();
+        }
+      }
+
+      class C2 : C1 {
+        var B : [D] string;
+
+        // ***** Unable to initialize omitted field *****
+        proc init(_D) {
+          super.init(_D);
+        }
+      }
+
+
+
+
+
+
+
+
+2. some users find some choices to be confusing and/or stylistically
+   unappealing.
+
+   The role of super.init() as the separator between phase 1 and
+   phase 2 remains unclear to some users.
+
+   Chapel allows a developer to omit an explicit call to a parent
+   initializer and rely on the compiler to insert this call.  It
+   is unclear whether it is more natural for an initalizer to
+   default to phase 1 or phase 2, i.e. whether the compiler
+   should insert the call at the start of the body or at the end
+   of the body, and it is hard to recall which choice has been made
+   for Chapel.
+
+   The Chapel language has been modified to disallow the potential to
+   support inheritance for records but we continue to rely on a call
+   to super.init() to signal that the type has been fully initialized.
+
+
+
+
+
+
+
+
+
+
+Additional concerns
++++++++++++++++++++
+
+3. The current implementation suppresses the compiler generated
+   initializers if the developer defines at least one
+   initializer. This ensures that a user does not unintentionally
+   rely on a compiler-generated all-fields initializer that fails
+   to meet the necessary semantics, but may require the type designer
+   to explicity write this initializer even in the simple cases.
+
+   The type designer also becomes responsible for the default
+   copy initializer if the type is a record.
+
+
+
+
+4. It may be unclear how to define a type specifier for a generic
+   type or how to define a minimal initializer.  For example what
+   is the signature for the compiler generated type constructor
+   in the following code fragment
+
+
+   .. code-block:: chapel
+
+      record MyR {
+        param flag  : bool;
+        var   value : int;
+
+        <a user defined initializer>
+      }
+
+      var myR : MyR( ... );
+
+
+   and what is the signature for a user-defined initializer that
+   will satisfy the variable declaration?  Should the answer be
+   different if this were a class instead of a record?
```

Comment:
I don't understand what either of the questions in this section are getting at.  If it's clear to everyone on the team, feel free to ignore me.